precompile/assetsbridge/bridge_out.go

```diff
@@ -0,0 +1,443 @@
+		assetsUnlocked, err = m.executeBitcoin(context, inputs)
+	}
+
+	if assetsUnlocked != nil {
+		err := context.EventEmitter().Emit(
+			NewAssetsUnlockedEvent(
+				assetsUnlocked.UnlockSequence.BigInt(),
+				assetsUnlocked.Recipient,
+				common.HexToAddress(assetsUnlocked.Token),
+				context.MsgSender(),
+				uint8(assetsUnlocked.Chain), //nolint:gosec // G115: Safe conversion, Chain is validated elsewhere
+				assetsUnlocked.Amount.BigInt(),
+			),
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to emit AssetsUnlocked event: [%w]", err)
+		}
+
+	}
+
+	return precompile.MethodOutputs{err == nil}, err
+}
+
+func (m *BridgeOutMethod) executeEthereum(
+	context *precompile.RunContext,
+	inputs *bridgeOutInputs,
+) (*bridgetypes.AssetsUnlockedEvent, error) {
+	var (
+		sdkCtx          = context.SdkCtx()
+		bridgeAddrBytes = common.HexToAddress(
+			evmtypes.AssetsBridgePrecompileAddress,
+		).Bytes()
+		spenderAddr = context.MsgSender()
+	)
+
+	sdkAmount, err := precompile.TypesConverter.BigInt.ToSDK(inputs.Amount)
+	if err != nil {
+		return nil, fmt.Errorf("unable to convert amount to sdk type: %v", err)
+	}
+
+	call, err := evmtypes.NewERC20BurnFromCall(
+		bridgeAddrBytes,
+		inputs.Token.Bytes(),
+		spenderAddr.Bytes(),
+		inputs.Amount,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create ERC20 burnFrom call: %w", err)
+	}
+
+	_, err = m.evmKeeper.ExecuteContractCall(sdkCtx, call)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute ERC20 burnFrom call: %w", err)
+	}
+
+	assetsUnlocked, err := m.bridgeKeeper.SaveAssetsUnlocked(
+		sdkCtx,
+		inputs.Token.Bytes(),
+		sdkAmount,
+		uint8(inputs.Chain),
+		inputs.Recipient,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send AssetsUnlocked to bridge: %w", err)
+	}
+
+	return assetsUnlocked, nil
+}
+
+func (m *BridgeOutMethod) executeBitcoin(
+	context *precompile.RunContext,
+	inputs *bridgeOutInputs,
+) (*bridgetypes.AssetsUnlockedEvent, error) {
+	// first check if the bridge is authorized to spend the funds.
+	bridgeAddrBytes := common.HexToAddress(
+		evmtypes.AssetsBridgePrecompileAddress,
+	).Bytes()
+	bridgeAddr := sdk.AccAddress(bridgeAddrBytes)
+	senderAddr := context.MsgSender()
+
+	authorization, expiration := m.authzKeeper.GetAuthorization(
+		context.SdkCtx(), bridgeAddr.Bytes(), senderAddr.Bytes(), SendMsgURL,
+	)
+	if authorization == nil {
+		return nil, fmt.Errorf("%s authorization type does not exist or is expired for address %s", SendMsgURL, senderAddr)
+	}
+
+	if expiration != nil && expiration.Before(context.SdkCtx().BlockTime()) {
+		return nil, fmt.Errorf("authorization expired at %v", expiration)
+	}
+
+	sdkAmount, err := precompile.TypesConverter.BigInt.ToSDK(inputs.Amount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert amount: [%w]", err)
+	}
+	coins := sdk.Coins{{Denom: evmtypes.DefaultEVMDenom, Amount: sdkAmount}}
+	msg := banktypes.NewMsgSend(bridgeAddrBytes, senderAddr.Bytes(), coins)
```

Comment:
I find the `executeBitcoin` function quite hard to follow.
For example, it seems like in this line:
```
msg := banktypes.NewMsgSend(bridgeAddrBytes, senderAddr.Bytes(), coins)
```
we are sending funds from the Bridge to the sender (the caller of the `bridgeOut` method).
So the sender gains funds. Shouldn't they lose funds if they are requesting bridging-out?

Short descriptions for `executeEthereum` and `executeBitcoin` would be helpful. For example they could briefly explain what the functions do with the focus on how the funds are moved/burned.